engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -197,101 +266,97 @@ public void render(DefaultRenderingProcess.StereoRenderState stereoRenderState)
         switch (stereoRenderState) {
             case MONO:
                 currentRenderingStage = WorldRenderingStage.DEFAULT;
+                isFirstRenderingStageForCurrentFrame = true;
                 break;
             case OCULUS_LEFT_EYE:
                 currentRenderingStage = WorldRenderingStage.OCULUS_LEFT_EYE;
                 playerCamera.updateFrustum();
+                isFirstRenderingStageForCurrentFrame = true;
                 break;
             case OCULUS_RIGHT_EYE:
                 currentRenderingStage = WorldRenderingStage.OCULUS_RIGHT_EYE;
                 playerCamera.updateFrustum();
+                isFirstRenderingStageForCurrentFrame = false;
                 break;
         }
 
-        if (stereoRenderState == DefaultRenderingProcess.StereoRenderState.MONO || stereoRenderState == DefaultRenderingProcess.StereoRenderState.OCULUS_LEFT_EYE) {
```

Comment:
AFAIK, at this stage there is no entry point for a pre-rendering phase. The update() method might seem like it but it's called every tick and in it there are things that I actually want to move out of there as they might get updated more than once per frame, unnecessarily. 

But I can certainly group various things in a single preRenderUpdate() method which is called first in the render() method. The firstRenderingStage check would occur in there.